pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -11,15 +11,13 @@ import (
 	"fmt"
```

Comment:
If we are going to call the new file  `v1alpha1_provider_clusterconfig.go`, then for consistency can we call `v1alpha1_provider.go` `v1alpha1_machine_provider.go`?

Reply:
Sure, I just wanted to show a part of it before potentially wasting time on full PR

Reply:
yes, and should we split `machine` part as well? and leave only core part in this file?